tools/shared.py

```diff
@@ -3006,6 +3006,9 @@ def get_subresource_location(path, data_uri=None):
     if data_uri is None:
       data_uri = Settings.SINGLE_FILE
     if data_uri:
+      # if the path does not exist, then there is no data to encode
+      if not os.path.exists(path):
+        return ''
```

Comment:
What kind of other files might not exist here?  Could this hide valid missing file errors?

Reply:
Other possible files are the asm.js code file, and the .mem file. Also the fastcomp emterpreter data file.

I agree there is a risk, if the file doesn't exist we assume it shouldn't exist. But it seems messier to add a param "allow non-existing" and use that in some places. Unless you have a better idea?

Reply:
I'll defer to you judgement here.  I don't know this part of the code too well.

Reply:
Ok, let's leave it as-is then. I think the minor risk (which would cause obvious breakage, so nothing that could be missed) is worth cleaner code.